classes/request.class.php

```diff
@@ -260,6 +260,13 @@ public function test_connection($apiurl, $apikey) {
             return json_encode($data);
         }
 
+        if (preg_match($validurlregexwithapi, $apiurl)) {
+            if ($this->logger) {
+                $this->logger->info('Stripping /api from Turnitin URL for connection test: ', array($apiurl));
+            }
+            $apiurl = str_replace("/api", '', $apiurl);
```

Comment:
I think all of this logic could be done with one regex (instead of two) and then selecting a group here that is everything except the optional (/api) at the end.

Reply:
I've added the optional /api to the original regex and removed the new one. I tried using a group in the regex e.g ?<uri_host> but when trying to select the group the PHP function preg_match was inconsistent with how it built the array when I had /api or not. Using str_replace seemed to be the better method to remove /api than writing more logic to handle the different arrays.